### ISSUE TITLE
GBrowse expression patterns don't appear to work

### ISSUE THREAD

**Issue by scottcain** [WormBase/website#3684]
In the "Expression Pattern" track, GBrowse is supposed to show a hover balloon with an expression cartoon where the glyph exists, however, no cartoon appears in any example I can find (20 or so examples on different chromosomes in C. elegans).  Part of the problem may be in the configuration: the config refers to this directory for images: /usr/local/wormbase/website-shared-files/images/expression/assembled/ but it doesn't exist on dev.  The configuration also references what is probably a cgi called gbrowse_popup, which appears to exist (http requests for it return an empty page) but doesn't appear to do anything.

This track is also supposed to use an "image" glyph, but this doesn't work, also because of the missing directory mentioned above.

If this were working, I could use the same images to implement similar functionality in JBrowse.

**Comment by tharris** [comment]
I've restored the directory - was previously served via proxy. 

gbrowse_popup was predecessor to balloons. Can you do this same functionality in the balloon callback?

**Comment by scottcain** [comment]
Should be able to; I'll work on it.

**Comment by scottcain** [comment]
Here is a simple demonstration that it works:

http://dev.wormbase.org:10004/tools/genome/gbrowse/elegans/?start=6381000;stop=6475999;ref=III;width=800;version=100;flip=0;grid=on;id=4b13ce56bef2c25b6ae36c4db5bd9d1b;l=EXPRESSION_PATTERNS%1EGENES%1ELOCI%3Aoverview

But I don't really know about how big the images in the track should be versus what information should be in the balloon.  Suggestions?

**Comment by tharris** [comment]
That looks good. There formerly was more detail provided by the gbrowse popup window about the nature of these images. 

But given the age of the images and the transition of the underlying DB, I think linking to the expo_pattern page would be sufficient.  I think you could also lose the balloon - just display it inline.

**Comment by scottcain** [comment]
I agree that the balloons should go away.  I'm at a little bit of a loss on the image sizes in the track though: there are a vareity of image sizes available on the server, but there doesn't appear to be any consistency about which ones I can always count on being available, as some have just the original, native size image and nothing else.  Currently the callback just goes through the resolutions I've found by clever perl one liner-fu: 100, 175, 200, 225, 250, 270, 300, and 400 pixels, which results in a variety of sizes of cartoons in the track.  I suppose if this were my data, I'd write a perl script to make 100 pixel options of each pattern and use that for GBrowse, do you have an objection to that, @tharris ?

**Comment by tharris** [comment]
Solution seems reasonable. The images are incredibly old. @chris-grove, @WenChenWormBase and others interested in expression, what are your thoughts on these?

**Comment by chris-grove** [comment]
@tharris  I don't know if people have a preference for these older cartoons. I generated Virtual Worm images for many expression patterns a while back, but I do need to get a per-release cycle of images going to stay up to date. The cartoons have the benefit that some features (e.g. small anatomy objects) are maybe easier to see than with the VW images, but there's more ambiguity and a need to interpret the cartoons. I can work on generating VW images and try to set it to some standard image size that works for GBrowse

**Comment by scottcain** [comment]
@chris-grove Are the virtual worm images the ones on the expression pages?  Those are pretty :-)  I imagine those would work jus as well--I don't know that anybody would make snap decisions based on what they see in gbrowse without following up, so a difficult to see pattern might just get somebody to click on it to see what they're missing.

**Comment by chris-grove** [comment]
@scottcain  Yes, those are the virtual worm images. OK, well you're welcome to use those for GBrowse if you'd like. I'm pretty sure they are all rendered to a standard size, but I'm not sure if it's the size you need. @tharris  could tell you where to find the images for display on the website. Otherwise I can create a new directory for you somewhere more appropriate

**Comment by draciti** [comment]
Scott, Todd,
just a thought. Could we display alongside the virtualworm renderings also the actual expression pictures?
http://www.wormbase.org/species/all/expr_pattern/Expr5285#0213--10
in this case there are 3. We could display just one and by clicking on it you will be redirected to the expression page
![screen shot 2015-03-23 at 4 33 56 pm](https://cloud.githubusercontent.com/assets/901243/6792894/89443c50-d17a-11e4-8500-9ffa6e72c690.png)
